autograder/handgrading/tests/test_views/test_handgrading_result_views.py

```diff
@@ -615,6 +616,91 @@ def test_student_delete_result_permission_denied(self) -> None:
+    @staticmethod
+    def format_group_data(group, handgrading_result: dict | None,
+                          has_handgradeable_submission: bool) -> dict:
+        """Format the group data for comparison with the response data."""
+        data = group.to_dict()
+        data['handgrading_result'] = handgrading_result
+        data['member_names'].sort()
+        data['has_handgradeable_submission'] = has_handgradeable_submission
+        return data
+
+    def generate_expected_data(self, group: ag_models.Group,
+                               rubric: hg_models.HandgradingRubric) -> dict:
+        """Generate the expected data base on submission status and handgrading result status."""
+        submission = self.create_submission(group)
+        match self.has_handgrading_result, self.submission_status:
+            # A finished_grading submission with a handgrading result
+            case True, _:
+                return self.generate_expected_data_with_handgrading_result(
+                    group, submission, rubric)
+            # A finished_grading submission without a handgrading result
+            # (has_handgradeable_submission would still be true in this case)
+            case False, ag_models.Submission.GradingStatus.finished_grading:
+                return self.format_group_data(group,
+                                              handgrading_result=None,
+                                              has_handgradeable_submission=True)
+            # A submission with any status other than
+            # finished_grading (and consequently no handgrading result either).
+            # has_handgradeable_submission would be false in this case.
+            case _:
+                return self.format_group_data(group,
+                                              handgrading_result=None,
+                                              has_handgradeable_submission=False)
+
+    def generate_expected_data_with_handgrading_result(
+            self, group: ag_models.Group, submission: ag_models.Submission,
+            rubric: hg_models.HandgradingRubric) -> dict:
+        """Generate the expected data for a group with a handgrading result."""
+        hg_result = hg_models.HandgradingResult.objects.validate_and_create(
+            submission=submission,
+            group=group,
+            handgrading_rubric=rubric
+        )
+        handgrading_data = {
+            'total_points': hg_result.total_points,
+            'total_points_possible': hg_result.total_points_possible,
+            'finished_grading': hg_result.finished_grading
+        }
+        match self.submission_status:
+            # Is handgradable when the submission has a finished grading status
+            case ag_models.Submission.GradingStatus.finished_grading:
+                return self.format_group_data(
+                    group, handgrading_data, has_handgradeable_submission=True)
+            case _:
+                return self.format_group_data(
```

Comment:
I think this case should be impossible (after applying the other changes I mentioned, I think this particular `match` statement can be removed).